### PR TITLE
fix(UI): Fix clear state button

### LINKED ISSUE #127: bug(state): Clear state button does not perform any action
**Issue:**
Like the title says the Clear state button does not perform any action

### PR DESCRIPTION
**Pull request by lgmarchi**
### Which issue(s) this PR fixes 🔨

- Resolve #127

### CODE REVIEW

File: src/lib/state/db/dbOperations.ts [hunk 1]
```diff
@@ -14,10 +14,22 @@ export async function initDB() {
...
-        const request = indexedDB.deleteDatabase("UplinkAppState")
-        request.onsuccess = () => resolve("")
-    })
+    return new Promise((resolve, reject) => {
+        console.log("Clearing state from DB");
+        const dbName = "UplinkAppState";
+        const request = indexedDB.deleteDatabase(dbName);
+
+        request.onerror = (event) => {
+            console.error(`Error deleting database ${dbName}`, event);
```
Comment (InfamousVague) [inline src/lib/state/db/dbOperations.ts:23]: Can we move these to use the logger so it's easier to disable dev logging

File: src/lib/state/db/dbOperations.ts [hunk 2]
```diff
@@ -14,10 +14,22 @@ export async function initDB() {
...
+            reject(event);
+        }
+
+        request.onblocked = () => {
+            console.warn(`Database ${dbName} deletion blocked`);
```
Comment (InfamousVague) [inline src/lib/state/db/dbOperations.ts:28]: same for this one

File: src/routes/settings/developer/+page.svelte [hunk 3]
```diff
@@ -31,7 +31,16 @@
...
-        <Button hook="button-clear-state" appearance={Appearance.Alt} on:click={_ => clearState()}>Clear State</Button>
+        <Button hook="button-clear-state" appearance={Appearance.Alt} 
+            on:click={async _ => {
+                    await clearState().then(() => {
+                        goto(Route.Unlock)
+                        setTimeout(() => {
+                            location.reload()
+                        }, 500)
+                    })
+                    .catch((error) => console.error('Error deleting database', error));
```
Comment (InfamousVague) [inline src/routes/settings/developer/+page.svelte:42]: and last one here

**Review by InfamousVague — COMMENTED** [review]
Small changes for the logging then looks good to me!

**Comment by luisecm** [comment]
We noticed the following bug:

1. With a user already having friends - Clicked on clear state button for a user already having friends
2. Was redirected to enter pin and then username (as it was a fresh one)
3. But then I go to Friends, and see that my "new user" has already a friend:

image

**Review by luisecm — APPROVED** [review]
Tested and passed

**Pull request merged**